### ISSUE TITLE
Unexpected accuracy output from tf.keras.Model.evaluate called on a saved and loaded model

### ISSUE THREAD

**Issue by ibeauregard** [tensorflow/tensorflow#43457]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 2.3.0
- Python version: 3.6.9
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 10.1
- GPU model and memory: N/A (run with x86_64 CPU)

**Describe the current behavior**
A model is trained for a multiclass classification task. Labels are a class vector. Method [evaluate](https://www.tensorflow.org/api_docs/python/tf/keras/Model#evaluate) called on the trained model outputs an accuracy of 0.9579. After saving the model and loading the saved model, calling method `evaluate` on the loaded model outputs an accuracy of 0.0997.

**Describe the expected behavior**
I would have expected the same accuracy output (0.9579) on the loaded model.

**Standalone code to reproduce the issue**
https://colab.research.google.com/drive/1L4vgn3H5PquUJEgBY3PCrlGEHSXd6Rtc

**Other info / logs**

**Comment by bhack** [comment]
It is a know issue and it is fixed in `tf-nightly` https://github.com/tensorflow/tensorflow/issues/42045#issuecomment-686769008
If you don't want to use nightly there is a workaround at https://github.com/tensorflow/tensorflow/issues/42045#issuecomment-674232499

**Comment by ibeauregard** [comment]
Thank you @bhack, and sorry I was not able to find that thread. Closing this issue.
